ext/ddtrace_profiling_native_extension/collectors_thread_context.c

```diff
@@ -1051,11 +1051,21 @@ static void trace_identifiers_for(struct thread_context_collector_state *state,
   }
 }
 
-static bool is_type_web(VALUE root_span_type) {
+// We only collect the resource for spans of types:
+// * 'web', for web requests
+// * proxy', used by the rack integration with request_queuing: true (e.g. also represents a web request)
+//
+// NOTE: Currently we're only interested in HTTP service endpoints. Over time, this list may be expanded.
+// Resources MUST NOT include personal identifiable information (PII); this should not be the case with
+// ddtrace integrations, but worth mentioning just in case :)
+static bool should_collect_resource(VALUE root_span_type) {
   ENFORCE_TYPE(root_span_type, T_STRING);
 
-  return RSTRING_LEN(root_span_type) == strlen("web") &&
-    (memcmp("web", StringValuePtr(root_span_type), strlen("web")) == 0);
+  int root_span_type_length = RSTRING_LEN(root_span_type);
+  const char *root_span_type_value = StringValuePtr(root_span_type);
+
+  return (root_span_type_length == strlen("web") && (memcmp("web", root_span_type_value, strlen("web")) == 0)) ||
```

Comment:
Since you are potentially doing 2 `strlen()` calls per string per call does it make sense to make these constants so we don't evaluate them every function call?

Reply:
Fair point

Reply:
Since this is `strlen` on a constant string, the compiler will optimize it away, even at `-O0`, see: https://godbolt.org/z/Wb351Monv :)